### ISSUE TITLE
DO NOT merge PRs with 'Squash and merge' button

### ISSUE THREAD

**Issue by peterlimg** [0chain/0chain#275]
This would squash all the commits history into one commit, it's ok to be used for your own PRs when the commits are trivial, but for others, squash their commits would lost all the origin commits. 

Use the default 'Merge pull request' button.

**Comment by TheTeaParty** [comment]
But squashing commits are good for main branches, it's easy to roll back, and it is easy to track changes specific to feature, otherwise we will have lots of unordered commits and lots of `Merge branch ... into ...` commits which are not informative at all.

**Comment by guruhubb** [comment]
Is unordered commit and Merge branch a problem other than appearance? 

I'd rather have an entire history of changes especially the important ones.  I agree with Peter that trivial commits should be squashed and decided by the person doing their own PRs, and one shouldn't squash other's PR commits.

**Comment by Sriep** [comment]
Failure to squash leaves a large number of noise commits that make it difficult to look through the history. My preference is to squash by default. 

One suggestion is to use [GitHub emoji](https://gist.github.com/parmentf/035de27d6ed1dce0b36a). You can use a GitHub emoji to indicate a commit you don't want to be squashed.

Letting everyone merge their own PRs is very dangerous. Many are shy about merging and never get around to it, others are too aggressive. Both tendencies lead to problems.

**Comment by guruhubb** [comment]
Ok makes sense.

**Comment by peterlimg** [comment]
Yes, good suggestion. We can left a emoji or message to indicate if we need to do a `squash and merge`. Even though, I agree with @guruhubb that we should keep the entire history of changes. We may have comments that describe the reason of the commits. Squashing may lose info, and contribute records. Though, the origin commits will not be shown in the history, but we can find them in the PR. The problem is we can not see it with local git commands, and if in some case, we migrate the code from the github to another platform, those squashed commits could not be migrated as well. 

For rollback @TheTeaParty , sometime, we may only need to rollback one or number of commits instead of the whole PR.

**Comment by guruhubb** [comment]
Ok. Lets do a message or emoji to squash, otherwise by default do NOT squash.

**Comment by Sriep** [comment]
Resolved.
